activesupport/lib/active_support/isolated_execution_state.rb

```diff
@@ -4,53 +4,58 @@
 
 module ActiveSupport
   module IsolatedExecutionState # :nodoc:
-    @isolation_level = :thread
+    @isolation_level = nil
 
     Thread.attr_accessor :active_support_execution_state
     Fiber.attr_accessor :active_support_execution_state
 
     class << self
       attr_reader :isolation_level
+      attr_accessor :scope
 
       def isolation_level=(level)
+        return if level == @isolation_level
+
         unless %i(thread fiber).include?(level)
           raise ArgumentError, "isolation_level must be `:thread` or `:fiber`, got: `#{level.inspect}`"
         end
 
-        if level != isolation_level
-          clear
-          singleton_class.alias_method(:current, "current_#{level}")
-          singleton_class.send(:private, :current)
-          @isolation_level = level
+        clear if @isolation_level
+
+        self.scope = case level
+                     when :thread; Thread
+                     when :fiber; Fiber
         end
```

Comment:
Ah, right. I believe Rubocop allows this variant a slightly tidier `end`:

```suggestion
        self.scope =
          case level
          when :thread; Thread
          when :fiber; Fiber
          end
```